merkledag/merkledag.go

```diff
@@ -371,27 +410,27 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, root *Node, set
 func fetchNodes(ctx context.Context, ds DAGService, in <-chan []key.Key, out chan<- *Node, errs chan<- error) {
 	defer close(out)
```

Comment:
(discussed offline; the answer is *no*: go guarantees sends+closes happen in order)